diagnostics/block.go

```diff
@@ -0,0 +1,150 @@
+package diagnostics
+
+import (
+	"container/list"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/ledgerwatch/erigon-lib/metrics"
+)
+
+type MilliSeconds time.Duration
+
+func (m MilliSeconds) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Duration(m).Milliseconds())
+}
+
+var (
+	averageHeaderDelayGauge         = metrics.NewGauge(`average_delay{type="header"}`)
+	averageBodyDelayGauge           = metrics.NewGauge(`average_delay{type="body"}`)
+	averageExecutionStartDelayGauge = metrics.NewGauge(`average_delay{type="execution_start"}`)
+	averageExectionEndDelayGauge    = metrics.NewGauge(`average_delay{type="execution_end"}`)
+	averageProductionDelayGauge     = metrics.NewGauge(`average_delay{type="production"}`)
```

Comment:
have you considered using prometheus Summary (or Histogram but Summary gives you some cool statistics for free) for this https://prometheus.io/docs/concepts/metric_types/ - we have funcs in erigon-lib/metrics for both - Histogram.ObserveDuration and Summary.ObserveDuration